test/Microsoft.AspNetCore.Server.KestrelTests/FrameTests.cs

```diff
@@ -1128,7 +1128,7 @@ public void TakeStartLineThrowsOnUnsupportedHttpVersion()
                 socketInput.IncomingData(requestLineBytes, 0, requestLineBytes.Length);
 
                 var exception = Assert.Throws<BadHttpRequestException>(() => frame.TakeStartLine(socketInput));
-                Assert.Equal("Unrecognized HTTP version.", exception.Message);
+                Assert.Equal("Unrecognized HTTP version: HTTP/1.2", exception.Message);
```

Comment:
Added. While adding it I realized it makes sense to limit the printed version to 9 chars instead of 8. Otherwise something like "HTTP/1.1a" appears as "HTTP/1.1..." in the log and that's not as helpful as showing at least one character that made the version "bad".